tests/integ/test_cte.py

```diff
@@ -331,14 +331,17 @@ def test_same_duplicate_subtree(session):
         assert count_number_of_ctes(df_result2.queries["queries"][-1]) == 3
 
 
-@pytest.mark.parametrize("mode", ["append", "overwrite", "errorifexists", "ignore"])
+@pytest.mark.parametrize(
+    "mode", ["append", "truncate", "overwrite", "errorifexists", "ignore"]
+)
 def test_save_as_table(session, mode):
     df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
-    expected_query_count = 1
     if mode == "append":
-        # append mode uses create table with insert
-        # TODO (SNOW-1703599): Show table query is executed twice when new compilation stage is enabled
-        expected_query_count = 4 if session._query_compilation_stage_enabled else 3
+        expected_query_count = 3
```

Comment:
let's add a comment here about what are the queries, it is probably 1 show table query, 1 create query, and one insert

Reply:
added